ffv1.md

```diff
@@ -0,0 +1,960 @@
+
+```
+log2_run[41]={JPEGLS.
+ 0, 0, 0, 0, 1, 1, 1, 1,
+ 2, 2, 2, 2, 3, 3, 3, 3,
+ 4, 4, 5, 5, 6, 6, 7, 7,
+ 8, 9,10,11,12,13,14,15,
+16,17,18,19,20,21,22,23,
+24,
+};
+
+if (run_count == 0 && run_mode == 1) {
+    if (get_bits1()) {
+        run_count = 1 << log2_run[run_index];
+        if (x + run_count <= w)
+            run_index++;
+    } else {
+        if (log2_run[run_index])
+            run_count = get_bits(log2_run[run_index]);
+        else
+            run_count = 0;
+        if (run_index)
+            run_index--;
+        run_mode = 2;
+    }
+}
+```
+
+[JPEGLS](#references)
+
+#### Level coding
+
+Level coding is identical to the normal difference coding with the exception that the 0 value is removed as it cannot occur:
+
+```
+    if(diff>0) diff--;
+    encode(diff);
+```
+
+Note, this is different from JPEG-LS, which doesn’t use prediction in run mode and uses a different encoding and context model for the last difference On a small set of test samples the use of prediction slightly improved the compression rate.
+
+# Bitstream
+
+|      |                                                        |
+|------|--------------------------------------------------------|
+| u(n) | unsigned big endian integer using n bits               |
+| sg   | Golomb Rice coded signed scalar symbol coded with the method described in [Huffman Coding Mode](#huffman-coding-mode) |
+| br   | Range coded boolean (1-bit) symbol with the method described in [Range binary values](#range-binary-values)  |
+| ur   | Range coded unsigned scalar symbol coded with the method described in [Range non binary values](#range-non-binary-values) |
+| sr   | Range coded signed scalar symbol coded with the method described in [Range non binary values](#range-non-binary-values) |
+
+The same context which is initialized to 128 is used for all fields in the header.
+
+Default values at the decoder initialization phase:
+
+**ConfigurationRecordIsPresent** is set to 0.
+
+## Configuration Record
+
+
+In the case of a bitstream with version >= 2, a configuration record is stored in the the underlying container, at the track header level.
+It contains the parameters used for all frames.
+The size of the configuration record, NumBytes, is supplied by the underlying container.
+
+|                                                          |      |
+|----------------------------------------------------------|-----:|
+| ConfigurationRecord( NumBytes ) {                        |      |
+|    ConfigurationRecordIsPresent = 1                      |      |
+|    Parameters( )                                         |      |
+|    while( remaining\_bits\_in\_bitstream( ) \> 32)       |      |
```

Comment:
Here and in a lot of places, please be consistent with white space. In this case, put a space after `32`.